packages/ckeditor5-image/src/imageinsert.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module image/imageinsert
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import ImageUpload from './imageupload';
+import ImageInsertUI from './imageinsert/imageinsertui';
+
+/**
+ * The image insert plugin.
+ *
+ * For a detailed overview, check the {@glink features/image-upload/image-upload Image upload feature}
+ * and {@glink features/image#inserting-images-via-source-url Insert images via source URL} documentation.
+ *
+ * This plugin does not do anything directly, but it loads a set of specific plugins
+ * to enable image uploading or inserting via implemented integrations:
+ *
+ * * {@link module:image/imageupload~ImageUpload}
+ * * {@link module:image/imageinsert/imageinsertui~ImageInsertUI},
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class ImageInsert extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get pluginName() {
+		return 'ImageInsert';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	static get requires() {
+		return [ ImageUpload, ImageInsertUI ];
```

Comment:
Doesn't it require only `ImageUploadUI`?

Reply:
Good question. I thought that when I "depend" on `ImageUpload` it pulls `ImageUploadUI` automatically, because the dependency chain looks like this: 

`ImageInsert` -> [
    `ImageUpload` -> [`ImageUploadUI`, `ImageUploadProgress`, `ImageUploadEditing`], 
    `ImageInsertUI`
]

So I don't need to rely on `ImageUploadProgress` and `ImageUploadEditing` here as well.

Reply:
Hmm... I guess **no**, because we rely on the whole plugin here, not only on its UI 🤔